iceoryx2/src/service/builder/blackboard.rs

```diff
@@ -246,33 +265,341 @@ impl<KeyType: ZeroCopySend + Debug, ServiceType: service::Service> Builder<KeyTy
+                let mut payload_size = 0;
+                if self.builder.internals.is_empty() {
+                    fail!(from self,  with BlackboardCreateError::NoEntriesProvided,
+                        "{} without entries. At least one key-value pair is required.", msg);
+                }
+                for i in &self.builder.internals {
+                    payload_size += i.internal_value_size + i.internal_value_alignment - 1;
+                }
+                let payload_shm = match <<ServiceType::BlackboardPayload as SharedMemory<
+                    iceoryx2_cal::shm_allocator::bump_allocator::BumpAllocator,
+                >>::Builder as NamedConceptBuilder<ServiceType::BlackboardPayload>>::new(
+                    &name
+                )
+                .config(&shm_config)
+                .has_ownership(false)
+                .size(payload_size)
+                .create(&iceoryx2_cal::shared_memory::bump_allocator::Config::default())
+                {
+                    Ok(v) => v,
+                    Err(_) => {
+                        fail!(from self, with BlackboardCreateError::ServiceInCorruptedState,
+                            "{} since the blackboard payload data segment could not be created. This could indicate a corrupted system.",
+                            msg);
+                    }
+                };
+
+                // create the management segment
+                let capacity = self.builder.internals.len();
+                let mgmt_config = blackboard_mgmt_config::<ServiceType, Mgmt<KeyType>>(
+                    self.builder.base.shared_node.config(),
+                );
+                let mgmt_storage = fail!(from self, when
+                    <ServiceType::BlackboardMgmt<Mgmt<KeyType>> as DynamicStorage<Mgmt<KeyType>,
+                    >>::Builder::new(&name)
+                        .config(&mgmt_config)
+                        .has_ownership(false)
+                        .supplementary_size(RelocatableFlatMap::<KeyType, usize>::const_memory_size(capacity)+RelocatableVec::<Entry>::const_memory_size(capacity))
+                        .initializer(|entry: &mut Mgmt<KeyType>, allocator: &mut BumpAllocator| {
+                            if unsafe {entry.map.init(allocator)}.is_err() || unsafe {entry.entries.init(allocator).is_err()} {
+                                return false
+                            }
+                            for i in 0..capacity {
+                                // write value passed to add() to payload_shm
+                                let mem = payload_shm.allocate(unsafe { Layout::from_size_align_unchecked(self.builder.internals[i].internal_value_size, self.builder.internals[i].internal_value_alignment) });
+                                if mem.is_err() {
+                                    error!("Writing the value to the blackboard data segment failed.");
```

Comment:
`error!` also supports the `from self` syntax. Could you please use it here for all error logs?

So adjust it to `error!(from self, "Writing ....`